cdci_data_analysis/analysis/renku_helper.py

```diff
@@ -147,7 +150,7 @@ def check_job_id_branch_is_present(repo, job_id):
     r = re.compile(f".*_{job_id}")
     filtered_list = list(filter(r.match, list_branches))
 
-    return len(filtered_list) == 1
```

Comment:
so this is only used for test, right?

why do we end up with >= 1 now, several sessions? Would several different sessions for the same jobid have the same exact content?

Reply:
> so this is only used for test, right?
> 
yes!

> why do we end up with >= 1 now, several sessions? Would several different sessions for the same jobid have the same exact content?
>

Because there could be the autosave ones:

![image](https://user-images.githubusercontent.com/3757833/205049234-92eea64a-6427-474c-b4a6-9169d075c7c7.png)

Reply:
could you actively distinguish autosaves, since they have distinctly different format? To avoid missing actual problem creating multiple branches - this should not happen.

Maybe only match branches with a more strict pattern?

In principle, users can also create similar branches.

Reply:
will address via #478